stdlib/public/Concurrency/Task.cpp

```diff
@@ -21,20 +21,129 @@
 #include "TaskPrivate.h"
 
 using namespace swift;
+using FutureFragment = AsyncTask::FutureFragment;
+
+size_t FutureFragment::fragmentSize(const Metadata *resultType) {
+  return storageOffset(resultType) + resultType->vw_size();
+}
+
+void FutureFragment::destroy() {
+  switch (status.load()) {
+  case Status::Executing:
+    assert(false && "destroying a task that never completed");
+
+  case Status::Success:
+    resultType->vw_destroy(getStoragePtr());
+    break;
+
+  case Status::Error:
+    swift_unknownObjectRelease(getStoragePtr());
```

Comment:
Please do not release an interior pointer into the future fragment. :)

Also, please use `swift_errorRelease`.

Reply:
`swift_errorRelease` is only defined when Objective-C interoperability is enabled. But yeah, the other thing is bad ;)